### PR TITLE
feat: implement param parsing of `SubmitRegionMigrationTaskHandler`

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Implement param parsing of SubmitRegionMigrationTaskHandler

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)
#2700

### CODE REVIEW

File: src/meta-srv/src/service/admin/region_migration.rs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    fn peer(&self, peer_id: u64) -> Option<Peer>;
+}
+
+/// The handler of submitting migration task.
+pub struct SubmitRegionMigrationTaskHandler {
+    // TODO(weny): waits for https://github.com/GreptimeTeam/greptimedb/pull/3014
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct SubmitRegionMigrationTaskParams {
```
Comment (fengjiachun) [inline src/meta-srv/src/service/admin/region_migration.rs:39]: Perhaps we could call it: SubmitRegionMigrationTaskRequest?

**Review by fengjiachun — APPROVED** [review]

File: src/meta-srv/src/service/admin/region_migration.rs (hunk 1 above)
Comment (WenyXu) [reply src/meta-srv/src/service/admin/region_migration.rs:39]: Yes, I think the `Params` is more precise. The `SubmitRegionMigrationTaskParams ` are parsed from the body of the request. IMO, a `Request` typically should contain the context of the request.

File: src/meta-srv/src/service/admin/region_migration.rs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        _: &str,
+        params: &HashMap<String, String>,
+    ) -> Result<http::Response<String>> {
+        let params = SubmitRegionMigrationTaskParams::try_from(params)?;
+
+        let response = self.handle_submit(params).await?;
+
+        http::Response::builder()
+            .status(http::StatusCode::OK)
+            .body(serde_json::to_string(&response).unwrap())
```
Comment (MichaelScofield) [inline src/meta-srv/src/service/admin/region_migration.rs:122]: Better returns error.

File: src/meta-srv/src/service/admin/region_migration.rs (hunk 1 above)
Comment (WenyXu) [reply src/meta-srv/src/service/admin/region_migration.rs]: I changed my mind. I changed it to `SubmitRegionMigrationTaskRequest `. We can have something like `Request< SubmitRegionMigrationTaskRequest>` in the future.

**Comment by WenyXu** [comment]
@MichaelScofield PTAL

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**